buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+import io.netty.util.internal.PlatformDependent;
+
+import java.nio.ByteOrder;
+
+/**
+ * All operations get and set as {@link ByteOrder#BIG_ENDIAN}.
+ */
+final class UnsafeByteBufUtil {
+
+    private static final boolean NATIVE_ORDER = ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN;
+    private static final boolean UNALIGNED = PlatformDependent.isUnaligned();
+
+    static byte getByte(long address) {
+        return PlatformDependent.getByte(address);
+    }
+
+    static short getShort(long address) {
+        if (UNALIGNED) {
+            short v = PlatformDependent.getShort(address);
+            return NATIVE_ORDER ? v : Short.reverseBytes(v);
+        }
+        return (short) (PlatformDependent.getByte(address) << 8 | PlatformDependent.getByte(address + 1) & 0xff);
+    }
+
+    static int getUnsignedMedium(long address) {
+        return (PlatformDependent.getByte(address) & 0xff) << 16 |
```

Comment:
Actually we don't need to check here, since `getShort` already does the right thing if `UNALIGNED`.